vector/src/test/java/im/vector/app/features/home/room/detail/timeline/factory/PollItemFactoryTest.kt

```diff
@@ -0,0 +1,282 @@
+import org.amshove.kluent.shouldBe
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.matrix.android.sdk.api.session.room.model.message.MessagePollContent
+import org.matrix.android.sdk.api.session.room.model.message.PollAnswer
+import org.matrix.android.sdk.api.session.room.model.message.PollCreationInfo
+import org.matrix.android.sdk.api.session.room.model.message.PollQuestion
+import org.matrix.android.sdk.api.session.room.model.message.PollType
+import org.matrix.android.sdk.api.session.room.send.SendState
+
+private val A_MESSAGE_INFORMATION_DATA = MessageInformationData(
+        eventId = "eventId",
+        senderId = "senderId",
+        ageLocalTS = 0,
+        avatarUrl = "",
+        sendState = SendState.SENT,
+        messageLayout = TimelineMessageLayout.Default(showAvatar = true, showDisplayName = true, showTimestamp = true),
+        reactionsSummary = ReactionsSummaryData(),
+        sentByMe = true,
+)
+
+private val A_POLL_RESPONSE_DATA = PollResponseData(
+        myVote = null,
+        votes = emptyMap(),
+)
+
+private val A_POLL_CONTENT = MessagePollContent(
+        unstablePollCreationInfo = PollCreationInfo(
+                question = PollQuestion(
+                        unstableQuestion = "What is your favourite coffee?"
+                ),
+                kind = PollType.UNDISCLOSED_UNSTABLE,
+                maxSelections = 1,
+                answers = listOf(
+                        PollAnswer(
+                                id = "5ef5f7b0-c9a1-49cf-a0b3-374729a43e76",
+                                unstableAnswer = "Double Espresso"
+                        ),
+                        PollAnswer(
+                                id = "ec1a4db0-46d8-4d7a-9bb6-d80724715938",
+                                unstableAnswer = "Macchiato"
+                        ),
+                        PollAnswer(
+                                id = "3677ca8e-061b-40ab-bffe-b22e4e88fcad",
+                                unstableAnswer = "Iced Coffee"
+                        ),
+                )
+        )
+)
+
+class PollItemFactoryTest {
+
+    private val testDispatcher = UnconfinedTestDispatcher()
+
+    @get:Rule
+    val mvRxTestRule = MvRxTestRule(
+            testDispatcher = testDispatcher // See https://github.com/airbnb/mavericks/issues/599
+    )
+
+    private lateinit var pollItemFactory: PollItemFactory
+
+    @Before
+    fun setup() {
+        // We are not going to test any UI related code
+        pollItemFactory = PollItemFactory(
+                stringProvider = mockk(),
+                avatarSizeProvider = mockk(),
+                colorProvider = mockk(),
+                dimensionConverter = mockk(),
+        )
+    }
+
+    @After
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun `given a sending poll state then PollState is Sending`() = runTest {
```

Comment:
`runTest` can be removed from all tests. It is only useful when testing `suspend` functions.

Reply:
Done.